tv_shows_app/src/components/features/reviews/ReviewItem/DeleteButton/DeleteButton.tsx

```diff
@@ -0,0 +1,86 @@
+import {
+  Button,
+  IconButton,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Spinner,
+  useDisclosure,
+} from '@chakra-ui/react';
+import { IRemoveReviewParams, IReview } from '../ReviewItem';
+import { DeleteIcon } from '@chakra-ui/icons';
+import useSWRMutation from 'swr/mutation';
+import { swrKeys } from '@/fetchers/swrKeys';
+import { deleteReviewMutator } from '@/fetchers/mutators';
+import { mutate } from 'swr';
+
+interface IDeleteButtonProps {
+  review: IReview;
+  onRemove: (reviewId: number) => void;
+}
+
+export const DeleteButton = ({ review, onRemove }: IDeleteButtonProps) => {
+  const { trigger } = useSWRMutation(
+    swrKeys.deleteReview(review.id),
+    deleteReviewMutator<IRemoveReviewParams>,
+    {
+      onSuccess: () => {
+        onRemove(review.id);
+      },
+    }
+  );
+  async function removeReview(params: IRemoveReviewParams) {
+    console.log(params);
+    await trigger(params);
+    mutate(`/api/shows/${review.show_id}`);
+  }
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const onClickHandler = () => {
+    console.log(review.id);
+    removeReview({
+      id: review.id,
+    });
+    onClose();
+  };
+  return (
+    <>
+      {review.user.id == -1 ||
+      localStorage.getItem('tv-shows-uid') == review.user?.id + '' ? (
+        <IconButton
+          data-testid="delete-button"
+          backgroundColor="orange.100"
+          _hover={{ backgroundColor: 'red.300' }}
+          marginLeft="auto"
+          aria-label="Delete review"
+          size="sm"
+          icon={<DeleteIcon />}
+          onClick={onOpen}
+        />
+      ) : (
+        <></>
+      )}
```

Comment:
Ovo ne zelim raditi. Moja komponenta se "uvijek" mora renderati.
Ovaj dio za conditional cu staviti u komponenti iznad (ReviewItem):
Ovu komponentu ne zanima tj. ne treba zanimati hoce li se prikazati ili ne, onda prikazuje svoj sadržaj kakav je (button + modal), a u komponenti inzad cemo onda reci u ovom slucaju nemoj prikazati delete button a u ovom pokazi.

Sa strane optimizacije to isto ima smisla jer ne zelim da se u kodu uopce pozove DeleteButton i alocira memorija i odradi sve unutar njega a nakon toga odluci hoce li se pokazati njegov sadržaj. Već ukoliko nije zadovoljen uvjet (pseudokod dolje) nemoj uopce prikazati komponentu (to uopce nece inicirati render niti ista od ovog button-a, basically ce ignorirati sve).

```typescript

isMyUser && <DeleteButton review={review} />
```